packages/tall-plugin-meta-refresh/src/index.ts

```diff
@@ -0,0 +1,42 @@
+import { IncomingMessage } from 'node:http'
+import * as htmlparser2 from 'htmlparser2'
+import { Follow, Stop } from 'tall'
+
+export async function metaRefreshPlugin(
+  url: URL,
+  response: IncomingMessage,
+  previous: Follow | Stop
+): Promise<Follow | Stop> {
+  let metaRefreshUrl: string | undefined
+
+  const parser = new htmlparser2.Parser({
+    onopentag(name, attributes) {
+      if (
+        !metaRefreshUrl &&
+        name === 'meta' &&
+        attributes['http-equiv'] === 'refresh'
+      ) {
+        const match = attributes.content.match(/url=['"]?([^'"]*)/i)
+        if (match) {
+          metaRefreshUrl = match[1]
+        }
+      }
+    }
+  })
+
+  for await (const chunk of response) {
+    if (metaRefreshUrl) {
+      // avoid parsing the entire response if we already found the meta refresh tag
+      break
+    }
+    parser.write(chunk.toString())
+  }
```

Comment:
Alternative for ending early using `parser.end()`

```suggestion
  const parser = new htmlparser2.Parser({
    onopentag(name, attributes) {
      if (name === 'meta' && attributes['http-equiv'] === 'refresh') {
        const match = attributes.content.match(/url=['"]?([^'"]*)/i)
        if (match) {
          metaRefreshUrl = match[1]
          parser.end()
        }
      }
    }
  })

  for await (const chunk of response) {
    // avoid parsing the entire response if we already found the meta refresh tag
    if (metaRefreshUrl) break
    parser.write(chunk.toString())
  }
```